lib/authorizer/index.js

```diff
@@ -110,7 +110,8 @@ authorizeRequest = function (request, done) {
 
     var clonedReq = new sdk.Request(request.toJSON()),
         auth = clonedReq.auth,
-        authInterface = createAuthInterface(auth),
+        authState = {}, // setting this as empty because dry run has no state
+        authInterface = createAuthInterface(auth, authState),
```

Comment:
We also need to handle state in `digest` auth.

Reply:
On. digest we are setting the user data
- nonce
- realm